notebooks/MNIST-torchvision.ipynb

```diff
@@ -0,0 +1,798 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x7f2d39578240>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plot_example(torch.stack(X_example), y_example)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Preparing a validation split\n",
+    "\n",
+    "skorch can split the data for us automatically but since we are using `Dataset`s for their lazy-loading property there is no way skorch can do a stratified split automatically without exploring the data completely first (which it doesn't). \n",
+    "\n",
+    "If we want skorch to do a validation split for us we need to retrieve the `y` values from the dataset and pass these values to `net.fit` later on:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 68,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "y_train = np.array([y for x, y in iter(mnist_train)])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Build Neural Network with PyTorch\n",
+    "\n",
+    "Simple, fully connected neural network with one hidden layer. Input layer has 784 dimensions (28x28), hidden layer has 98 (= 784 / 8) and output layer 10 neurons, representing digits 0 - 9."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 25,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from torch import nn\n",
+    "import torch.nn.functional as F"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 26,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "device = 'cuda' if torch.cuda.is_available() else 'cpu'\n",
+    "mnist_flat_dim = 28 * 28"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A Neural network in PyTorch's framework."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 64,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ClassifierModule(nn.Module):\n",
+    "    def __init__(\n",
+    "            self,\n",
+    "            input_dim=mnist_flat_dim,\n",
+    "            hidden_dim=98,\n",
+    "            output_dim=10,\n",
+    "            dropout=0.5,\n",
+    "    ):\n",
+    "        super(ClassifierModule, self).__init__()\n",
+    "        self.dropout = nn.Dropout(dropout)\n",
+    "\n",
+    "        self.hidden = nn.Linear(input_dim, hidden_dim)\n",
+    "        self.output = nn.Linear(hidden_dim, output_dim)\n",
+    "\n",
+    "    def forward(self, X, **kwargs):\n",
+    "        X = X.reshape(-1, self.hidden.in_features)\n",
+    "        X = F.relu(self.hidden(X))\n",
+    "        X = self.dropout(X)\n",
+    "        X = F.softmax(self.output(X), dim=-1)\n",
+    "        return X"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "skorch allows to use PyTorch's networks in the SciKit-Learn setting:"
```

Comment:
I find "in the SciKit-Learn setting" a bit vague, what about "with an sklearn API" or something along the lines.